domain/src/test/java/org/oppia/domain/analytics/AnalyticsControllerTest.kt

```diff
@@ -0,0 +1,271 @@
+      ApplicationProvider.getApplicationContext(),
+      TEST_TIMESTAMP,
+      EventAction.EVENT_ACTION_UNSPECIFIED,
+      analyticsController.createExplorationContext(
+        TEST_TOPIC_ID,
+        TEST_STORY_ID,
+        TEST_EXPLORATION_ID
+      )
+    )
+
+    assertThat(fakeEventLogger.getMostRecentEvent().actionName)
+      .isEqualTo(EventAction.EVENT_ACTION_UNSPECIFIED)
+    assertThat(fakeEventLogger.getMostRecentEvent().timestamp).isEqualTo(TEST_TIMESTAMP)
+    assertThat(fakeEventLogger.getMostRecentEvent().priority).isEqualTo(Priority.OPTIONAL)
+    assertThat(fakeEventLogger.getMostRecentEvent().context.activityContextCase)
+      .isEqualTo(EXPLORATION_CONTEXT)
+  }
+
+  @Test
+  fun testController_LogClickEvent_withTopicContext_checkLogsEvent() {
+    fakeEventLogger.clearAllEvents()
+
+    analyticsController.logClickEvent(
+      ApplicationProvider.getApplicationContext(),
+      TEST_TIMESTAMP,
+      EventAction.EVENT_ACTION_UNSPECIFIED,
+      analyticsController.createTopicContext(TEST_TOPIC_ID)
+    )
+
+    assertThat(fakeEventLogger.getMostRecentEvent().actionName)
+      .isEqualTo(EventAction.EVENT_ACTION_UNSPECIFIED)
+    assertThat(fakeEventLogger.getMostRecentEvent().timestamp).isEqualTo(TEST_TIMESTAMP)
+    assertThat(fakeEventLogger.getMostRecentEvent().priority).isEqualTo(Priority.OPTIONAL)
+    assertThat(fakeEventLogger.getMostRecentEvent().context.activityContextCase)
+      .isEqualTo(TOPIC_CONTEXT)
+  }
+
+  @Test
+  fun testController_LogClickEvent_withStoryContext_checkLogsEvent() {
+    fakeEventLogger.clearAllEvents()
+
+    analyticsController.logClickEvent(
+      ApplicationProvider.getApplicationContext(),
+      TEST_TIMESTAMP,
+      EventAction.EVENT_ACTION_UNSPECIFIED,
+      analyticsController.createStoryContext(TEST_TOPIC_ID, TEST_STORY_ID)
+    )
+
+    assertThat(fakeEventLogger.getMostRecentEvent().actionName)
+      .isEqualTo(EventAction.EVENT_ACTION_UNSPECIFIED)
+    assertThat(fakeEventLogger.getMostRecentEvent().timestamp).isEqualTo(TEST_TIMESTAMP)
+    assertThat(fakeEventLogger.getMostRecentEvent().priority).isEqualTo(Priority.OPTIONAL)
+    assertThat(fakeEventLogger.getMostRecentEvent().context.activityContextCase)
+      .isEqualTo(STORY_CONTEXT)
+  }
+
+  @Test
+  fun testController_LogClickEvent_withNoContext_checkLogsEvent() {
+    fakeEventLogger.clearAllEvents()
+
+    analyticsController.logClickEvent(
+      ApplicationProvider.getApplicationContext(),
+      TEST_TIMESTAMP,
+      EventAction.EVENT_ACTION_UNSPECIFIED,
+      null
+    )
+
+    assertThat(fakeEventLogger.getMostRecentEvent().actionName)
+      .isEqualTo(EventAction.EVENT_ACTION_UNSPECIFIED)
+    assertThat(fakeEventLogger.getMostRecentEvent().timestamp).isEqualTo(TEST_TIMESTAMP)
+    assertThat(fakeEventLogger.getMostRecentEvent().priority).isEqualTo(Priority.OPTIONAL)
+    assertThat(fakeEventLogger.getMostRecentEvent().context.activityContextCase)
+      .isEqualTo(ACTIVITYCONTEXT_NOT_SET)
+  }
```

Comment:
Can you add tests for the createExplorationContext/ createQuestionContext/ createTopicContext/ createStoryContext methods as well?

Reply:
Umm.. is it not enough ? I mean we're actually logging the event using these functions and then later on checking if the context has been set to what we intended to or not ?

Reply:
Done.